backend/core/llm_chain.py

```
@@ -347,26 +358,34 @@ async def stream_with_fallback(
             logger.info(f"Attempting to stream response using {llm_name.title()}...")
             qa_chain = create_qa_chain(llm_instance)
-            async for chunk in qa_chain.astream({"input": user_input, "context": unique_docs}):
-                yield chunk
-                full_response_chunks.append(chunk)
-            stream_successful = True
-            logger.info(f"Successfully streamed response with {llm_name.title()}.")
-            break
+
+            async def llm_stream():
+                full_response_chunks = []
+                async for chunk in qa_chain.astream({"input": user_input, "context": unique_docs}):
```

Ruff (0.12.2)

Function definition does not bind loop variable `qa_chain`

(B023)